crm_location/models/crm_lead.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2015 Antiun Ingenieria - Endika Iglesias <endikaig@antiun.com>
+# Copyright 2017 Tecnativa - Luis Martínez
+# Copyright 2019 Tecnativa - Alexandre Díaz
+# License AGPL-3 - See https://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    @api.onchange("location_id")
+    def on_change_city(self):
+        if self.location_id:
```

Comment:
As fields have been changed to computed readonly False, could you override _compute_partner_address_values method instead?